lib/buffer.js

```diff
@@ -549,33 +549,10 @@ Buffer.prototype.toJSON = function() {
 };
 
 
-// TODO(trevnorris): currently works like Array.prototype.slice(), which
-// doesn't follow the new standard for throwing on out of range indexes.
```

Comment:
Thanks for noticing this @ChALkeR 

Looking back, I put this in over 2 years ago (3a2f273b) when I was attempting to be more strict with how arguments were handled. Now that we've been forced to use typed arrays it's safe to say that the original intention of this comment is dead. It's fine to leave it removed.

Oop, and just saw that it was added back. That's cool to. Maybe we should get some feedback about whether it _should_ throw. I dunno anymore.